payment-in-stablecoins/README.md

```diff
@@ -0,0 +1,133 @@
+
+## Technical Specification
+
+### Overview
+
+#### Interaction flow
+
+1. A proposal is created, where the DAO wants to pay the builder $100K:
+   1. Assuming ETH/USD is $1500, and a volatility buffer of 2x
+   2. The proposal transaction would be:
+      `NounsStablecoinPayments.mintIOUs(recipient, 100_000)`
+      where `msg.value = NounsStablecoinPayments.ethNeeded(100_000, 2)`
+2. Upon proposal execution:
+   1. Proposal recipient is minted 100K NOU tokens
+   2. 133.33 ETH is sent from the treasury to NounsStablecoinPayments
+3. Arbitrageurs sell USD to NounsStablecoinPayments in exchange for ETH:
+   1. They call `NounsStablecoinPayments.sellStablecoin(100_000)` (or smaller amounts across multiple txs)
+      1. Expected to happen when `NounsStablecoinPayments.price()` is better than other DEX prices
+   2. NounsStablecoinPayments's USD balance grows by $100K, and an appropriate ETH balance is spent
+4. Proposal recipient redeems USD in exchange for NOU tokens:
+   1. They call `NounsStablecoinPayments.redeem()`
+   2. 100K NOU tokens are burned, and 100K USD is transferred from NounsStablecoinPayments to recipient
+
+#### `NOU` contract
+
+A new ERC20 token contract:
+
+- Inherits from OpenZeppelin's `AccessControlEnumerable`
+  - Assigns the `ADMIN_ROLE` to an `admin` provided in the constructor; admin can administer its own role and all other roles
+- `function mint(address recipient, uint amount)`
+  - Mints `amount` tokens to `recipient`
+  - Permits only senders of role `MINTER_ROLE`
+- `function burn(address from, uint amount)`
+  - Burns `amount` tokens from `from`'s account
+  - Permits only senders of role `BURNER_ROLE`
+
+#### `NounsStablecoinPayments` contract
+
+A new contract that helps the DAO acquire stablecoins by buying them from bots for a margin, and helps proposal recipients redeem stablecoin payments.
+
+##### Contract state variables
+
+Immutable:
+
+- `iouToken` the address of the NOU token ERC20 contract
+- `stablecoin` the address of the ERC20 token to use as USD, e.g. USDC or DAI
+- `oracle` the Chainlink Stablecoin/ETH oracle address
+
+Mutable:
+
+- `baselineStablecoinAmount` how much stablecoin should the contract buy on top of outstanding IOU, to serve as a buffer than can expedite recipient redemption
+- `botIncentiveBPs` the basis points to add on top of the oracle price to further incentivize bots to sell USD to this contract
+- `admin` the address of the admin that can withdraw this contract's balances
+
+##### View functions
+
+- `function stablecoinAmountNeeded()` returns the amount of USD the DAO needs to buy, should be the same as `iouToken.totalSupply() - stablecoin.balanceOf(this) + baselineStablecoinAmount`
+- `function price()` returns the amount stablecoin this contract is asking for in exchange for one ETH, taking into account oracle pricing and the additional bot incentive factor: `Chainlink(eth-usd.data.eth).latestRoundData().answer * (10_000 + botIncentiveBPs) / 10_000`
```

Comment:
Assuming `eth-usd.data.eth` is only used as an example, but the Chainlink oracle should be specific to `stablecoin` (`DAI/ETH`, `USDC/ETH`).

Reply:
agreed, clarified this is just an example